cordova-lib/src/plugman/registry/registry.js

```diff
@@ -114,19 +113,11 @@ function initThenLoadSettingsWithRestore(promises) {
 }
 
 /**
-* @param {Array} with one element - the plugin id or "id@version"
+* @param {string} plugin - the plugin id or "id@version"
 * @return {Promise.<string>} Promised path to fetched package.
 */
 function fetchPlugin(plugin) {
-    return initThenLoadSettingsWithRestore(function () {
-        events.emit('log', 'Fetching plugin "' + plugin + '" via npm');
-        return Q.ninvoke(npm.commands, 'cache', ['add', plugin])
-        .then(function (info) {
-            var unpack = require('../../util/unpack');
-            var pluginDir = path.resolve(npm.cache, info.name, info.version, 'package');
-            // Unpack the plugin that was added to the cache (CB-8154)
-            var package_tgz = path.resolve(npm.cache, info.name, info.version, 'package.tgz');
-            return unpack.unpackTgz(package_tgz, pluginDir);
-        });
-    });
+    events.emit('log', 'Fetching plugin "' + plugin + '" via npm');
+    var pluginSplit = plugin.split('@');
+    return npmhelper.fetchPackage(pluginSplit[0], pluginSplit[1]);
```

Comment:
Looks like this will not work for scoped NPM packages (like `@scope/cordova-plugin-camera@2.0.0`). `pluginSplit[0]` and `pluginSplit[1]` would become `''` and `'scope/cordova-plugin-camera'` in this case.